src/main/java/software/amazon/cloudformation/proxy/AmazonWebServicesClientProxy.java

```diff
@@ -422,15 +408,10 @@ public Logger getLogger() {
                                         return ProgressEvent.failed(model, context, HandlerErrorCode.NotStabilized,
                                             "Exceeded attempts to wait");
                                     }
-                                    long remainingTime = getRemainingTimeInMillis();
-                                    long localWait = next.toMillis() + 2 * elapsed + 100;
-                                    if (remainingTime > localWait) {
-                                        loggerProxy.log("Waiting for " + next.getSeconds() + " for call " + callGraph);
-                                        Uninterruptibles.sleepUninterruptibly(next.getSeconds(), TimeUnit.SECONDS);
-                                        continue;
+                                    event = AmazonWebServicesClientProxy.this.waitStrategy.await(elapsed, next, context, model);
```

Comment:
nit:  Would be good to use a different variable as  it's about whether to continue.  

```
final Optional<ProgressEvent> completeLocalExecution = AmazonWebServicesClientProxy.this.waitStrategy.await(elapsed, next, context, model);
if (completeLocalExecution.present()) {
 return completeLocalExecution.get();
}
```